src/dag/get.js

```diff
@@ -22,31 +24,20 @@ module.exports = (send) => {
     options = options || {}
     path = path || ''
 
-    if (CID.isCID(cid)) {
-      cid = cid.toBaseEncodedString()
+    try {
+      const res = ipfsPath(cid)
+      cid = res.cid
+      path = res.path || path
+    } catch (err) {
+      return callback(err)
     }
 
-    waterfall([
-      cb => {
-        send({
-          path: 'dag/resolve',
-          args: cid + '/' + path,
-          qs: options
-        }, cb)
-      },
-      (resolved, cb) => {
-        block(send).get(new CID(resolved['Cid']['/']), (err, ipfsBlock) => {
-          cb(err, ipfsBlock, resolved['RemPath'])
-        })
-      },
-      (ipfsBlock, path, cb) => {
-        if (ipfsBlock.cid.codec === 'dag-cbor') {
-          dagCBOR.resolver.resolve(ipfsBlock.data, path, cb)
-        }
-        if (ipfsBlock.cid.codec === 'dag-pb') {
-          dagPB.resolver.resolve(ipfsBlock.data, path, cb)
-        }
-      }
-    ], callback)
+    IPLDResolver.inMemory((err, ipld) => {
+      if (err) return callback(explain(err, 'failed to create IPLD resolver'))
+      ipld.support.rm(dagPB.resolver.multicodec)
```

Comment:
Can you please add a comment here, why this is needed? It's clear to me, but I don't think it's clear when you read the code for the first time.